### DISCUSSION TITLE
$wrapNodes to TextNode

### DISCUSSION THREAD

**Discussion by DeadMonstr** [facebook/lexical#5611] [Q&A]
Hello everyone, I need $wrapNodes func to be worked on TextNode.

`<span className="text-bold">Hello</span>` to be
 `<span className="myClass" >
<span className="text-bold">Hello</span>
</span>`

or

`<span className="text-bold myClass">Hello</span>`

when I added class with 
`const selectedItem = editor.getElementByKey(nodes[0].getKey())`
`selectedItem.classList.add("myClass")`

$generateHtmlFromNodes is not generated span with my class

Thank you for answer, sorry my english :-)

**Comment by 2wheeh** [comment]
1. `<span className="text-bold">Hello</span>` to be
`<span className="myClass" > <span className="text-bold">Hello</span> </span>` :

    You can replace the `TextNode` with `CustomTextNode` (docs: [node-replacement](https://lexical.dev/docs/concepts/node-replacement ))

2. `<span className="text-bold myClass">Hello</span>` :

    This way seems easier. You can inject `myClass` to certain Node using `theme` of initialConfig. (docs: [theming](https://lexical.dev/docs/getting-started/theming))

**Reply by 2wheeh** [reply]
If you are asking `$wrapNodes` implement to wrap TextNode on certain action, this wouldn't help you though 🤔.

**Reply by DeadMonstr** [reply]
2 : How to add class to selected textNode, Can you show some examples with theme

**Reply by 2wheeh** [reply]
> 2 : How to add class to selected textNode, Can you show some examples with theme

After I see this comment, It seems 1,2 both are useless for your use case

**Reply by 2wheeh** [reply]
For me to think of it, `<span className="myClass">TextNode</span>` this should be a new Node (i.e. MyNode)
so `$wrapNodes` does just change selected TextNodes to MyNodes.

**Reply by DeadMonstr** [reply]
thanks for answer but it's not
